dockstore-integration-testing/src/test/java/io/dockstore/client/cli/ExtendedMetricsTRSOpenApiIT.java

```diff
@@ -230,6 +230,7 @@ void testSubmitMetricsData() throws IOException {
         tool.setRegistryString(Registry.QUAY_IO.getDockerPath());
         tool.setDefaultVersion("symbolic.v1");
         tool.setDefaultCWLTestParameterFile("/examples/cgpmap/bamOut/bam_input.json");
+        tool.setType("DockstoreTool");
```

Comment:
This is dumb, it looks like the generated openapi classes do not set the type correctly for `DockstoreTool` so we need to manually set it for a couple tests

Reply:
Could we change the `publish` implementation on line 235/236 to set the `type`, and therefore be backwards compatible? Presumably it wouldn't de-serialize though?

Reply:
Unfortunately, the method called on line 235/236 is also generated code.